yarn-project/end-to-end/src/bench/client_flows/utils.ts

```diff
@@ -0,0 +1,267 @@
+    assert(GATE_TYPES.includes(gateType as GateType), `Gate type ${gateType} is not recognized`);
+    return { [gateType]: gateSize };
+  });
+
+  assert(traceSizes.length === GATE_TYPES.length, 'Decoded trace sizes do not match expected amount of gate types');
+  return traceSizes.reduce((acc, curr) => ({ ...acc, ...curr }), {}) as StructuredTrace;
+}
+
+function getMaxMemory(logs: Log[]): number {
+  const candidateLogs = logs.slice(0, 100).filter(log => /\(mem: .*MiB\)/.test(log.message));
+  const usage = candidateLogs.map(log => {
+    const memStr = log ? log.message.slice(log.message.indexOf('(mem: ') + 6, log.message.indexOf('MiB') - 3) : '';
+    return memStr ? parseInt(memStr) : 0;
+  });
+  return Math.max(...usage);
+}
+
+export function generateBenchmark(
+  flow: string,
+  logs: Log[],
+  timings: ProvingTimings | SimulationTimings,
+  privateExecutionSteps: PrivateExecutionStep[],
+  proverType: ProverType,
+  error: string | undefined,
+): ClientFlowBenchmark {
+  const minimumTrace = getMinimumTrace(logs);
+  const maxMemory = getMaxMemory(logs);
+
+  const steps = privateExecutionSteps.reduce<Step[]>((acc, step, i) => {
+    const previousAccGateCount = i === 0 ? 0 : acc[i - 1].accGateCount!;
+    return [
+      ...acc,
+      {
+        functionName: step.functionName,
+        gateCount: step.gateCount,
+        accGateCount: previousAccGateCount + step.gateCount!,
+        time: step.timings.witgen,
+      },
+    ];
+  }, []);
+  const totalGateCount = steps[steps.length - 1].accGateCount;
+  return {
+    name: flow,
+    timings: {
+      total: timings.total,
+      sync: timings.sync!,
+      proving: (timings as ProvingTimings).proving,
+      unaccounted: timings.unaccounted,
+      witgen: timings.perFunction.reduce((acc, fn) => acc + fn.time, 0),
+    },
+    maxMemory,
+    proverType,
+    minimumTrace,
+    totalGateCount: totalGateCount!,
+    steps,
+    error,
+  };
+}
+
+export function convertProfileToGHBenchmark(benchmark: ClientFlowBenchmark): GithubActionBenchmarkResult[] {
+  return [
+    {
+      name: 'witgen',
+      value: benchmark.timings.witgen,
+      unit: 'ms',
+    },
+    {
+      name: 'proving',
+      value: benchmark.timings.proving!,
+      unit: 'ms',
+    },
+    {
+      name: 'total',
+      value: benchmark.timings.total,
+      unit: 'ms',
+    },
+    {
+      name: 'sync',
+      value: benchmark.timings.sync!,
+      unit: 'ms',
+    },
+    {
+      name: 'unaccounted',
+      value: benchmark.timings.unaccounted,
+      unit: 'ms',
+    },
+    {
+      name: 'max_memory',
+      value: benchmark.maxMemory,
+      unit: 'MiB',
+    },
+    {
+      name: 'total_gate_count',
+      value: benchmark.totalGateCount,
+      unit: 'gates',
+    },
+  ];
+}
+
+export async function captureProfile(
+  label: string,
+  interaction: ContractFunctionInteraction | DeployMethod,
+  opts?: Omit<ProfileMethodOptions & DeployOptions, 'profileMode'>,
```

Comment:
Why didn't you like the explicit long name it has before? Not like this is used tons